runtime/gateway.go

```diff
@@ -311,18 +322,31 @@ func (gateway *Gateway) setupMetrics(config *StaticConfig) error {
 	// TODO: decide what default tags we want...
 	defaultTags := map[string]string{}
 
-	prefix := config.MustGetString("metrics.tally.service") +
-		".production.all-workers"
-	flushIntervalConfig := config.MustGetInt("metrics.tally.flushInterval")
+	// create metrics root scope
+	gateway.MetricScope, gateway.metricScopeCloser = tally.NewRootScope(
+		tally.ScopeOptions{
+			Tags:           defaultTags,
+			Prefix:         service + ".production.all-workers",
+			CachedReporter: metricsBackend,
+			Separator:      tally.DefaultSeparator,
+		},
+		time.Duration(config.MustGetInt("metrics.tally.flushInterval"))*time.Millisecond,
+	)
 
-	scope, closer := tally.NewRootScope(tally.ScopeOptions{
-		Tags:           defaultTags,
-		Prefix:         prefix,
-		CachedReporter: metricsBackend,
-		Separator:      tally.DefaultSeparator,
-	}, time.Duration(flushIntervalConfig)*time.Millisecond)
-	gateway.MetricScope = scope
-	gateway.metricScopeCloser = closer
+	// start collecting runtime metrics
+	gateway.runtimeMetrics = StartRuntimeMetricsCollector(
+		RuntimeMetricsOptions{
+			EnableCPUMetrics: config.MustGetBoolean("metrics.runtime.enableCPUMetrics"),
+			EnableMemMetrics: config.MustGetBoolean("metrics.runtime.enableMemMetrics"),
+			EnableGCMetrics:  config.MustGetBoolean("metrics.runtime.enableGCMetrics"),
+			CollectInterval:  time.Duration(config.MustGetInt("metrics.runtime.collectIntervalMs")) * time.Millisecond,
+		},
+		gateway.MetricScope.SubScope("runtime").Tagged(map[string]string{
+			"host":    hostname,
```

Comment:
As discussed, creating one RootScope for `service + ".production"` and subscopes for `all-workers` and `per-workers`.